timemachine/fe/atom_mapping.py

```diff
@@ -119,3 +124,64 @@ def get_core_by_mcs(mol_a, mol_b, query, threshold=0.5):
         raise AtomMappingError(f"not all mapped atoms are within {threshold:.3f}Å of each other")
 
     return core
+
+
+def get_core_with_alignment(
+    mol_a: Chem.Mol,
+    mol_b: Chem.Mol,
+    threshold: float = 2.0,
+    n_steps: int = 200,
+    k: float = 10000,
+    ff: Optional[Forcefield] = None,
+) -> NDArray:
+    """Selects a core betwen two molecules, by finding an initial core then aligning based on the core.
+
+    Parameters
+    ----------
+    mol_a: RDKit Mol
+
+    mol_b: RDKit Mol
+
+    threshold: float
+        Threshold between atoms in angstroms
+
+    n_steps: float
+        number of steps to run for alignment
+
+    ff: Forcefield or None
+        Forcefield to use for alignment, defaults to DEFAULT_FF forcefield if None
+
+    Returns
+    -------
+    core : np.ndarray of ints, shape (n_MCS, 2)
+
+    Notes
+    -----
+    * Warning! The initial core can contain an incorrect mapping, in that case the
+        core returned will be the same as running mcs followed by get_core_by_mcs.
+    """
+    # Copy mols so that when we change coordinates doesn't corrupt inputs
+    a_copy = deepcopy(mol_a)
+    b_copy = deepcopy(mol_b)
+
+    if ff is None:
+        ff = Forcefield.load_from_file(DEFAULT_FF)
+
+    # Remove hydrogens
+    a_without_hs = Chem.RemoveHs(deepcopy(a_copy))
+    b_without_hs = Chem.RemoveHs(deepcopy(b_copy))
+
+    def setup_core(mol_a, mol_b):
+        result = mcs(mol_a, mol_b, threshold=threshold)
+        query_mol = Chem.MolFromSmarts(result.smartsString)
+        core = get_core_by_mcs(mol_a, mol_b, query_mol, threshold=threshold)
+        return core
+
+    heavy_atom_core = setup_core(a_without_hs, b_without_hs)
+
+    conf_a, conf_b = align_mols_by_core(mol_a, mol_b, heavy_atom_core, ff, n_steps=n_steps, k=k)
+    set_romol_conf(a_copy, conf_a)
+    set_romol_conf(b_copy, conf_b)
+
+    core = setup_core(a_copy, b_copy)
+    return core
```

Comment:
It's useful for debugging to return the smarts string too

Reply:
Done in [4e0b6e3](https://github.com/proteneer/timemachine/pull/813/commits/4e0b6e3ae5aa5b4df619e835571c541cbc972436)